services/diversity/src/services/richness.js

```diff
@@ -12,105 +12,112 @@ module.exports = (RichnessPersistence, restAPI) => {
      *
      * @returns {Object[]} Number of inferred and observed species for the desired group.
      */
-    getNumberSpecies: async (areaType, areaId, group = 'all') => {
-      const data = [
-        {
-          id: 'total',
-          inferred: 40,
-          observed: 4,
-          region_observed: 110,
-          region_inferred: 130,
-        },
-        {
-          id: 'endemic',
-          inferred: 20,
-          observed: 25,
-          region_observed: 120,
-          region_inferred: 100,
-        },
-        {
-          id: 'invasive',
-          inferred: 20,
-          observed: 10,
-          region_observed: 90,
-          region_inferred: 80,
-        },
-        {
-          id: 'threatened',
-          inferred: 15,
-          observed: 20,
-          region_observed: 90,
-          region_inferred: 100,
-        },
-      ];
+    getNumberOfSpecies: async (areaType, areaId, group = 'all') => {
+      const promises = [];
       switch (group) {
         case 'total':
-          return [data[0]];
+          promises.unshift(RichnessPersistence.findTotalNumberOfSpecies(areaType, areaId));
+          break;
         case 'endemic':
-          return [data[1]];
+          promises.unshift(RichnessPersistence.findEndemicNumberOfSpecies(areaType, areaId));
+          break;
         case 'invasive':
-          return [data[2]];
+          promises.unshift(RichnessPersistence.findInvasiveNumberOfSpecies(areaType, areaId));
+          break;
         case 'threatened':
-          return [data[3]];
+          promises.unshift(RichnessPersistence.findThreatenedNumberOfSpecies(areaType, areaId));
+          break;
         case 'all':
+          promises.unshift(
+            RichnessPersistence.findTotalNumberOfSpecies(areaType, areaId),
+            RichnessPersistence.findEndemicNumberOfSpecies(areaType, areaId),
+            RichnessPersistence.findInvasiveNumberOfSpecies(areaType, areaId),
+            RichnessPersistence.findThreatenedNumberOfSpecies(areaType, areaId),
+          );
+          break;
         default:
-          return data;
+          throw new Error('Data for  doesn\'t exists for the given group');
```

Comment:
Fix the message. There is a extra `for` before `doesn't`

Reply:
changed.